### PR TITLE
Port to ocidir 0.3, containers-image-proxy 0.7

### PR DESCRIPTION
**Pull request by jeckersb**
Signed-off-by: John Eckersberg <jeckersb@redhat.com>

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -17,3 +17,7 @@ debug = true
 codegen-units = 1
 inherits = "release"
 lto = "yes"
+
+[patch.crates-io]
```
Comment (jeckersb) [inline Cargo.toml:21]: Note the explicit patch to align with the containers-image-proxy change here, need to coordinate moving these in tandem.

**Comment by cgwalters** [comment]
Ah sorry, my porting to `thiserror` broke this one. I think we may just need a few `.map_err(Into::into)`?

**Comment by jeckersb** [comment]
> Ah sorry, my porting to `thiserror` broke this one. I think we may just need a few `.map_err(Into::into)`?

Holy cow is that an absolutely cryptic compiler error now :laughing:

File: lib/src/container/unencapsulate.rs [hunk 2]
```diff
@@ -289,19 +290,17 @@ pub(crate) async fn fetch_layer_decompress<'a>(
...
             size = layer.size();
             media_type = layer.media_type();
-            (blob, driver) = proxy
-                .get_blob(img, layer.digest().as_str(), size as u64)
-                .await?;
+            (blob, driver) = proxy.get_blob(img, layer.digest(), size).await?;
         }
     };
 
+    let driver = async { driver.await.map_err(Into::into) };
```
Comment (jeckersb) [inline lib/src/container/unencapsulate.rs:302]: seems to be enough to accomodate the `thiserror` change, took me longer than I would care to admit to understand the compiler error before.

Comment (cgwalters) [reply lib/src/container/unencapsulate.rs:302]: I struggled a lot with this when I was learning Rust for sure. The thing that is subtle is that the `?` operator automatically also does `Into::into`, which means in cases where you're changing the error type (like this) but *not* using `?` (which happens in most cases) you suddenly have to think about these conversions.

**Comment by cgwalters** [comment]
Ah, test run hit https://github.com/ostreedev/ostree-rs-ext/issues/664

**Review by cgwalters — APPROVED** [review]
LGTM but we also need to remember this is a semver-breaking change for us, so we'll need to bump the version for the next release.

**Comment by cgwalters** [comment]
Fallout in https://github.com/containers/bootc/issues/800

I have yanked version 0.15.0.

**Pull request merged**